pyk/src/pyk/proof/proof.py

```diff
@@ -273,7 +277,7 @@ def lines(self) -> list[str]:
         return CompositeSummary([BaseSummary(self.id, self.status), *subproofs_summaries])
 
     @abstractmethod
-    def get_steps(self) -> Iterable[PS]: ...
+    def get_steps(self, kcfg_semantics: KCFGSemantics) -> Iterable[PS]: ...
```

Comment:
I'm hoping this is fixed now without creating other problems. I'm still waiting on these changes to finish testing on kontrol. To fix it I needed to include in the `APRProofStep` the prior loops cache and a tuple of `Nodes` which represent the path leading up to the current node. I am a bit worried this may unintentionally revert the performance fix that introduced prior loop caching: https://github.com/runtimeverification/pyk/pull/922.

@ovatman, would you be able to help me figure out what the requirements for the caching were and if they are still satisfied with this change?